### ISSUE TITLE
concurrent_skip_list: how to handle zero-sized variables (compare/allocator)?

### ISSUE THREAD

**Issue by igchor** [pmem/libpmemobj-cpp#755]
For example, if we change something like that:
```
allocator alloc;
int x;
```

to:

```
allocator alloc;
char x[4];
```

this will remove explicit padding between alloc and x (if sizeof(alloc) == 0).

I propose to add explicit padding in structures like comparator and alloctor (to 1 byte).

**Comment by vinser52** [comment]
I think we also should align concurrent_hash_map design in the same way.

**Comment by vinser52** [comment]
regarding your example, you changing the layout of the structure which contains allocator but not allocator itself. How padding inside allocator would help?

**Comment by igchor** [comment]
You're right, I've changed the description. I was thinking about aligning the allocator/comparator to 8 bytes (in that case, there would be no implicit padding between allocator and following datatype if it requires alignement <= 8). But probably a better approach is to add explicit padding in the hash_map/map itself (and just test it using layout tests).
